app/views/users/show.html.erb

```diff
@@ -1,3 +1,19 @@
-<h1>Single user</h1>
-<h2>Bio</h2>
-<h3>Posts</h3>
+<% @users = User.all; %>
+<% @posts = Post.all; %>
+<% current_user = @users.find_by(id: params[:id]) %>
+<% @posts_per_user = current_user.posts %>
+<h2><%= current_user.name %>'s page</h2>
+<div class="body">
+  <%= render partial: "user", locals: { current_user: current_user } %>
+  <div class="bio">
+    <h2>Bio</h2>
+    <p><%= current_user.bio %></p>
+  </div>
+  <h2>Last three posts</h2>
+  <% @posts_per_user.first(3).each do |post| %>
```

Comment:
- [ ] please avoid using logic in the views, you previously created a method called `last_3_posts` in your user model to handle this.